File: 4th Semester/Design and programming environments/Proiect C-Sharp Client-Server Sockets/proto/ProtoUtils.cs
using AppNetworking;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proto
{
    static class ProtoUtils
    {
        public static App.Protocol.Request createLoginRequest(AppModel.Utilizator user)
        {
            //new proto.User {Id = user.Id, Passwd = user.Password};
            App.Protocol.Utilizator userDTO = new App.Protocol.Utilizator { Id = user.id, Parola = user.parola, User = user.user };
            App.Protocol.Request request = new App.Protocol.Request { Type = App.Protocol.Request.Types.Type.Login, User = userDTO };

            return request;
        }

        public static App.Protocol.Request createLogoutRequest(AppModel.Utilizator user)
        {
            App.Protocol.Utilizator userDTO = new App.Protocol.Utilizator { Id = user.id };
            App.Protocol.Request request = new App.Protocol.Request { Type = App.Protocol.Request.Types.Type.Logout, User = userDTO };
            return request;
        }

        internal static App.Protocol.Response createGetNrParticipanti(List<int> list)
        {
            App.Protocol.Response response = new App.Protocol.Response
            {
                Type = App.Protocol.Response.Types.Type.GetNrParticipanti
            };
            foreach (int nr in list)
            {
                response.NrParticipnati.Add(nr);
            }

            return response;
        }

        /*public static Request createSendMesssageRequest(chat.model.Message message)
{
   proto.Message messageDTO = new proto.Message
   {
       SenderId = message.Sender.Id,
       ReceiverId = message.Receiver.Id,
       Text = message.Text
   };

   ChatRequest request = new ChatRequest { Type = ChatRequest.Types.Type.SendMessage, Message = messageDTO };
   return request;
}


public static ChatRequest createLoggedFriendsRequest(chat.model.User user)
{
   proto.User userDTO = new proto.User { Id = user.Id };
   ChatRequest request = new ChatRequest
   {
       Type = ChatRequest.Types.Type.GetLoggedFriends,
       User = userDTO
   };
   return request;
}*/


        public static App.Protocol.Response createOkResponse()
        {
            App.Protocol.Response response = new App.Protocol.Response { Type = App.Protocol.Response.Types.Type.Ok };
            return response;
        }


        public static App.Protocol.Response createErrorResponse(String text)
        {
            App.Protocol.Response response = new App.Protocol.Response
            {
                Type = App.Protocol.Response.Types.Type.Error,
                Error = text
            };
            return response;
        }

        internal static App.Protocol.Response createGetProbeResponse(List<AppModel.Proba> list)
        {
            App.Protocol.Response response = new App.Protocol.Response
            {
                Type = App.Protocol.Response.Types.Type.GetProbe
            };
            foreach (AppModel.Proba proba in list)
            {
                App.Protocol.Proba userDTO = new App.Protocol.Proba { Distanta = proba.distanta, Stil = proba.stil.ToString(), Id = proba.id };

                response.Probe.Add(userDTO);
            }

            return response;
        }

        internal static App.Protocol.Response createGetParticipantiDupaProbaResponse(List<AppModel.Participant> list)
        {
            App.Protocol.Response response = new App.Protocol.Response
            {
                Type = App.Protocol.Response.Types.Type.GetProbe
            };
            foreach (AppModel.Participant participant in list)
            {
                App.Protocol.Participant userDTO = new App.Protocol.Participant { Nume = participant.nume, Varsta = participant.varsta, Id = participant.id };

                response.Participanti.Add(userDTO);
            }

            return response;
        }

        /*public static Response createFriendLoggedInResponse(chat.model.User user)
        {
            proto.User userDTO = new proto.User { Id = user.Id };
            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.FriendLoggedIn, User = userDTO };
            return response;
        }

        public static ChatResponse createFriendLoggedOutResponse(chat.model.User user)
        {
            proto.User userDTO = new proto.User { Id = user.Id };
            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.FriendLoggedOut, User = userDTO };
            return response;
        }

        public static ChatResponse createNewMessageResponse(chat.model.Message message)
        {
            proto.Message messageDTO = new proto.Message
            {
                SenderId = message.Sender.Id,
                ReceiverId = message.Receiver.Id,
                Text = message.Text
            };

            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.NewMessage, Message = messageDTO };
            return response;
        }


        public static ChatResponse createLoggedFriendsResponse(chat.model.User[] users)
        {
            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.GetLoggedFriends
            };
            foreach (chat.model.User user in users)
            {
                proto.User userDTO = new proto.User { Id = user.Id };

                response.Friends.Add(userDTO);
            }

            return response;
        }*/

        public static String getError(App.Protocol.Response response)
        {
            String errorMessage = response.Error;
            return errorMessage;
        }

        internal static AppModel.Participant getParticipantInscriere(App.Protocol.Request request)
        {
            AppModel.Participant participant = new AppModel.Participant(request.ParticipnatNou.Varsta, request.ParticipnatNou.Nume);
            participant.id = request.ParticipnatNou.Id;
            return participant;
        }

        internal static List<AppModel.Proba> getProbeInscriere(App.Protocol.Request request)
        {
            List<AppModel.Proba> list = new List<AppModel.Proba>();
            foreach(App.Protocol.Proba p in request.ListProbe)
            {
                AppModel.Proba proba = new AppModel.Proba(p.Distanta, (AppModel.Stil)Enum.Parse(typeof(AppModel.Stil), p.Stil));
                list.Add(proba);
            }
            return list;
            
        }

        internal static App.Protocol.Response createGetProbeDupaParticipantResponse(List<AppModel.Proba> list)
        {
            App.Protocol.Response response = new App.Protocol.Response
            {
                Type = App.Protocol.Response.Types.Type.GetProbe
            };
            foreach (AppModel.Proba proba in list)
            {
                App.Protocol.Proba userDTO = new App.Protocol.Proba { Distanta = proba.distanta, Stil = proba.stil.ToString(), Id = proba.id };

                response.ProbeParticipant.Add(userDTO);
            }

            return response;
        }

        /* public static App.Protocol.Utilizator getUser(App.Protocol.Response response)
         {
             AppModel.Utilizator user = new AppModel.Utilizator(response);
             return user;
         }*/

        public static AppModel.Utilizator getUser(App.Protocol.Request request)
        {
            AppModel.Utilizator user = new AppModel.Utilizator(request.User.User, request.User.Parola);
            user.id = request.User.Id;
            return user;
        }

        public static AppModel.Proba getProba(App.Protocol.Request request)
        {
            AppModel.Proba proba = new AppModel.Proba(request.Proba.Distanta, (AppModel.Stil)Enum.Parse(typeof(AppModel.Stil), request.Proba.Stil));
            proba.id = request.Proba.Id;
            return proba;
        }

        public static AppModel.Participant getParticipant(App.Protocol.Request request)
        {
            AppModel.Participant participant = new AppModel.Participant(request.Participant.Varsta, request.Participant.Nume);
            participant.id = request.Participant.Id;
            return participant;
        }


        /*public static chat.model.Message getMessage(ChatResponse response)
        {
            chat.model.User sender = new chat.model.User(response.Message.SenderId);
            chat.model.User receiver = new chat.model.User(response.Message.ReceiverId);
            chat.model.Message message = new chat.model.Message(sender, receiver, response.Message.Text);
            return message;
        }

        public static chat.model.User[] getFriends(ChatResponse response)
        {
            chat.model.User[] friends = new chat.model.User[response.Friends.Count];
            for (int i = 0; i < response.Friends.Count; i++)
            {
                chat.model.User user = new chat.model.User(response.Friends[i].Id);
                friends[i] = user;
            }
            return friends;
        }

        public static chat.model.User getUser(ChatRequest request)
        {
            chat.model.User user = new chat.model.User(request.User.Id);
            user.Password = request.User.Passwd;
            return user;
        }
        public static chat.model.Message getMessage(ChatRequest request)
        {
            chat.model.User sender = new chat.model.User(request.Message.SenderId);
            chat.model.User receiver = new chat.model.User(request.Message.ReceiverId);
            chat.model.Message message = new chat.model.Message(sender, receiver, request.Message.Text);
            return message;
        }*/

    }
}

File: 4th Semester/Design and programming environments/Project Java Client-Server/Networking/src/main/java/protobuffprotocol/ProtoUtils.java
package protobuffprotocol;

import domain.Participant;
import domain.Proba;
import domain.Utilizator;
import domain.enums.Stil;

import java.util.ArrayList;
import java.util.List;

public class ProtoUtils {
    public static ChatProtobufs.Request createLoginRequest(Utilizator user) {
        ChatProtobufs.Utilizator userDTO = ChatProtobufs.Utilizator.newBuilder().setUser(user.getUser()).setParola(user.getParola()).build();
        ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Login)
                .setUser(userDTO).build();
        return request;
    }

    public static ChatProtobufs.Request createLogoutRequest(Utilizator user) {
        ChatProtobufs.Utilizator userDTO = ChatProtobufs.Utilizator.newBuilder().setUser(user.getUser()).build();
        ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Logout)
                .setUser(userDTO).build();
        return request;
    }

    public static ChatProtobufs.Request createGetProbeRequest() {
        ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Get_Probe)
                .build();
        return request;
    }

    public static ChatProtobufs.Request createGetParticipantiDupaProbaRequest(Proba proba) {
        ChatProtobufs.Proba probaDTO = ChatProtobufs.Proba.newBuilder().setDistanta(proba.getDistanta()).setId(proba.getId()).setStil(proba.getStil().toString()).build();
        ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Get_Participanti_Dupa_Probe)
                .setProba(probaDTO).build();
        return request;
    }

    public static ChatProtobufs.Request createGetProbeDupaParticipantRequest(Participant participant) {
        ChatProtobufs.Participant participantDTO = ChatProtobufs.Participant.newBuilder().setId(participant.getId()).setNume(participant.getNume()).setVarsta(participant.getVarsta()).build();
        ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Get_Probe_Dupa_Participant)
                .setParticipant(participantDTO).build();
        return request;
    }

    public static ChatProtobufs.Request createInscrieRequest(Participant participant, Proba[] probe) {
        ChatProtobufs.Participant participantDTO = ChatProtobufs.Participant.newBuilder().setNume(participant.getNume()).setVarsta(participant.getVarsta()).build();
        ChatProtobufs.Request.Builder request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Inscrie);
        request.setParticipnatNou(participantDTO);
        for(int i=0;i<probe.length;i++){
            ChatProtobufs.Proba probaDTO = ChatProtobufs.Proba.newBuilder().setDistanta(probe[i].getDistanta()).setId(probe[i].getId()).setStil(probe[i].getStil().toString()).build();
            request.addListProbe(probaDTO);
        }
        //ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Inscrie)
        //        .setParticipnatNou(participantDTO).setListProbe(listDTO).build();
        return request.build();
    }

    public static ChatProtobufs.Request createGetNrParticipantiRequest() {
        ChatProtobufs.Request request = ChatProtobufs.Request.newBuilder().setType(ChatProtobufs.Request.Type.Get_Nr_Participanti)
                .build();
        return request;
    }

//    public static ChatProtobufs.Request createSendMesssageRequest(Message message){
//        ChatProtobufs.Message messageDTO=ChatProtobufs.Message.newBuilder().
//                setSenderId(message.getSender().getId())
//                .setReceiverId(message.getReceiver().getId())
//                .setText(message.getText()).build();
//        ChatProtobufs.ChatRequest request= ChatProtobufs.ChatRequest.newBuilder()
//                .setType(ChatProtobufs.ChatRequest.Type.SendMessage)
//                .setMessage(messageDTO).build();
//        return request;
//    }
//
//    public static ChatProtobufs.Request createLoggedFriendsRequest(User user){
//        ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//        ChatProtobufs.ChatRequest request= ChatProtobufs.ChatRequest.newBuilder()
//                .setType(ChatProtobufs.ChatRequest.Type.GetLoggedFriends)
//                .setUser(userDTO).build();
//        return request;
//    }


    public static ChatProtobufs.Response createOkResponse() {
        ChatProtobufs.Response response = ChatProtobufs.Response.newBuilder()
                .setType(ChatProtobufs.Response.Type.Ok).build();
        return response;
    }

    public static ChatProtobufs.Response createErrorResponse(String text) {
        ChatProtobufs.Response response = ChatProtobufs.Response.newBuilder()
                .setType(ChatProtobufs.Response.Type.Error)
                .setError(text).build();
        return response;
    }

    public static ChatProtobufs.Response createGetProbeResponse(Proba[] probas) {

        ChatProtobufs.Response.Builder response = ChatProtobufs.Response.newBuilder()
                .setType(ChatProtobufs.Response.Type.Get_Probe);
        for (Proba proba : probas) {
            ChatProtobufs.Proba probaDTO = ChatProtobufs.Proba.newBuilder().setId(proba.getId()).setDistanta(proba.getDistanta()).setStil(proba.getStil().toString()).build();
            response.addProbe(probaDTO);
        }

        return response.build();
    }

    public static ChatProtobufs.Response createGetParticipnatiDupaProbaResponse(Participant[] participants) {
        ChatProtobufs.Response.Builder response = ChatProtobufs.Response.newBuilder()
                .setType(ChatProtobufs.Response.Type.Get_Participanti_Dupa_Probe);
        for (Participant participant : participants) {
            ChatProtobufs.Participant participantDTO = ChatProtobufs.Participant.newBuilder().setId(participant.getId()).setVarsta(participant.getVarsta()).setNume(participant.getNume()).build();
            response.addParticipanti(participantDTO);
        }

        return response.build();
    }

    public static ChatProtobufs.Response createGetProbeDupaParticipantResponse(Proba[] probas) {
        ChatProtobufs.Response.Builder response = ChatProtobufs.Response.newBuilder()
                .setType(ChatProtobufs.Response.Type.Get_Probe);
        for (Proba proba : probas) {
            ChatProtobufs.Proba probaDTO = ChatProtobufs.Proba.newBuilder().setId(proba.getId()).setDistanta(proba.getDistanta()).setStil(proba.getStil().toString()).build();
            response.addProbe(probaDTO);
        }

        return response.build();
    }

//    public static ChatProtobufs.Response createFriendLoggedInResponse(User user){
//        ChatProtobufs.Utilizator userDTO=ChatProtobufs.Utilizator.newBuilder().setId(user.getId()).build();
//
//        ChatProtobufs.Response response=ChatProtobufs.Response.newBuilder()
//                .setType(ChatProtobufs.Response.Type.FriendLoggedIn)
//                .setUser(userDTO).build();
//        return response;
//    }
//
//    public static ChatProtobufs.Response createFriendLoggedOutResponse(User user){
//        ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//
//        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.FriendLoggedOut)
//                .setUser(userDTO).build();
//        return response;
//    }
//    public static ChatProtobufs.Response createNewMessageResponse(Message message){
//        ChatProtobufs.Message messageDTO=ChatProtobufs.Message.newBuilder()
//                .setSenderId(message.getSender().getId())
//                .setReceiverId(message.getReceiver().getId())
//                .setText(message.getText())
//                .build();
//
//        ChatProtobufs.Response response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.NewMessage)
//                .setMessage(messageDTO).build();
//        return response;
//    }
//
//    public static ChatProtobufs.Response createLoggedFriendsResponse(User[] users){
//        ChatProtobufs.ChatResponse.Builder response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.GetLoggedFriends);
//        for (User user: users){
//            ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//            response.addFriends(userDTO);
//        }
//
//        return response.build();
//    }

    public static String getError(ChatProtobufs.Response response) {
        String errorMessage = response.getError();
        return errorMessage;
    }

    public static Proba[] getProbe(ChatProtobufs.Response response) {
        Proba[] friends = new Proba[response.getProbeCount()];
        for (int i = 0; i < response.getProbeCount(); i++) {
            ChatProtobufs.Proba userDTO = response.getProbe(i);
            Proba user = new Proba();
            user.setId(userDTO.getId());
            user.setDistanta(userDTO.getDistanta());
            user.setStil(Stil.valueOf(userDTO.getStil()));
            friends[i] = user;
        }
        return friends;
    }

    public static Participant[] getParticipanti(ChatProtobufs.Response response) {
        Participant[] friends = new Participant[response.getParticipantiCount()];
        for (int i = 0; i < response.getParticipantiCount(); i++) {
            ChatProtobufs.Participant userDTO = response.getParticipanti(i);
            Participant user = new Participant();
            user.setId(userDTO.getId());
            user.setNume(userDTO.getNume());
            user.setVarsta(userDTO.getVarsta());
            friends[i] = user;
        }
        return friends;
    }

    public static Proba[] getProbeParticipant(ChatProtobufs.Response response) {
        Proba[] friends = new Proba[response.getProbeParticipantCount()];
        for (int i = 0; i < response.getProbeParticipantCount(); i++) {
            ChatProtobufs.Proba userDTO = response.getProbeParticipant(i);
            Proba user = new Proba();
            user.setId(userDTO.getId());
            user.setDistanta(userDTO.getDistanta());
            user.setStil(Stil.valueOf(userDTO.getStil()));
            friends[i] = user;
        }
        return friends;
    }

    public static Utilizator getUser(ChatProtobufs.Request request) {
        Utilizator user = new Utilizator();
        user.setId(request.getUser().getId());
        user.setUser(request.getUser().getUser());
        user.setParola(request.getUser().getParola());
        return user;
    }

    public static List<Integer> getNrParticipanti(ChatProtobufs.Response response) {
        return response.getNrParticipnatiList();
    }

//    public static User getUser(ChatProtobufs.Response response){
//        User user=new User();
//        user.setId(response.getUser().getId());
//        return user;
//    }
//
//    public static Message getMessage(ChatProtobufs.ChatResponse response){
//        User sender=new User();
//        sender.setId(response.getMessage().getSenderId());
//        User receiver=new User();
//        receiver.setId(response.getMessage().getReceiverId());
//        Message message=new Message(sender,response.getMessage().getText(), receiver);
//        return message;
//    }
//
//    public static User[] getFriends(ChatProtobufs.ChatResponse response){
//        User[] friends=new User[response.getFriendsCount()];
//        for(int i=0;i<response.getFriendsCount();i++){
//            ChatProtobufs.User userDTO=response.getFriends(i);
//            User user=new User();
//            user.setId(userDTO.getId());
//            friends[i]=user;
//        }
//        return friends;
//    }
//    public static User getUser(ChatProtobufs.ChatRequest request){
//        User user=new User();
//        user.setId(request.getUser().getId());
//        user.setPasswd(request.getUser().getPasswd());
//        return user;
//    }
//
//    public static Message getMessage(ChatProtobufs.ChatRequest request){
//        User sender=new User();
//        sender.setId(request.getMessage().getSenderId());
//        User receiver=new User();
//        receiver.setId(request.getMessage().getReceiverId());
//        Message message=new Message(sender,request.getMessage().getText(), receiver);
//        return message;
//    }

}
